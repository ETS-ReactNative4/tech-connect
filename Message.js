import React, { Component } from 'react'
import { StyleSheet, View, Image, Text, TouchableOpacity } from 'react-native'

export default class Message extends Component {
  constructor() {
    super()
    this.state = {
      showMessage: false
    }
  }

  render() {
    const sentDate = new Date(this.props.message.created_at).toUTCString().split(' ').slice(0, 4).join(' ')
    const meetingDate = new Date(this.props.message.meeting_date).toUTCString().split(' ').slice(0, 4).join(' ')
    const time = new Date(this.props.message.meeting_date).toUTCString().split(' ')[4]
    
    return (
      <View style={ styles.messageContainer }>
        <Text style={ styles.date }>{ sentDate }</Text>
        <TouchableOpacity style={ styles.messageInfoContainer } onPress={() => this.setState({showMessage: !this.state.showMessage})}>
          <View style={ styles.imageContainer }>
            <Image source={ require('./profile-pic.jpeg') } style={ styles.userPic }/>
          </View>
          <View style={ styles.messageInfo }>
            <Text style={ styles.name }>{ this.props.message.receiver }</Text>
            <Text style={ styles.message }>{ this.props.message.status }</Text>
            {
              !this.state.showMessage ? null : <Text style={ styles.messageBody } >{ this.props.message.receiver } has confirmed a meeting with you at { this.props.message.meeting_location } on { meetingDate } at { time }</Text>
            }
          </View>
        </TouchableOpacity>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  messageContainer: {
    display: 'flex',
    flexDirection: 'row',
    padding: 20,
    borderBottomWidth: 0.5,
    borderBottomColor: 'gray',
  },
  messageInfoContainer: {
    display: 'flex',
    flexDirection: 'row',
    maxWidth: '100%',
    marginRight: 10
  },
  imageContainer: {
    shadowOffset: {  width: 2,  height: 2 },
    shadowRadius: 4,
    shadowColor: 'black',
    shadowOpacity: .2,
  },
  userPic: {
    height: 50,
    width: 50,
    borderRadius: 25
  },
  date: {
    position: 'absolute',
    right: 20,
    top: 20
  },
  name: {
    fontSize: 18
  },
  message: {
    marginTop: 5,
    color: 'gray',
  },
  messageBody: {
    marginRight: 10
  },
  messageInfo: {
    marginLeft: 10,
    marginRight: 30,
    marginTop: 5
  }
})