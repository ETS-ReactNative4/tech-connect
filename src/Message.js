import React, { Component } from 'react'
import { StyleSheet, View, Image, Text, TouchableOpacity } from 'react-native'
import { connect } from 'react-redux';


export class Message extends Component {
  constructor() {
    super()
    this.state = {
      showMessage: false
    }
  }

  render() {
    const { message, user } = this.props
    const connection = (user.name === message.receiver.name) ? message.sender.name :  message.receiver.name
    const sentDate = new Date(message.created_at).toUTCString().split(' ').slice(0, 4).join(' ')
    const meetingDate = new Date(message.meeting_date).toUTCString().split(' ').slice(0, 4).join(' ')
    const time = new Date(message.meeting_date).toUTCString().split(' ')[4]
    let messageText
    let messagePhoto
    let messageEnd = (message.meeting_location !== 'N/A') ? `at ${message.meeting_location} on ${meetingDate} at ${time}` : null

    if (message.receiver.name === user.name) {
      messageText = <Text style={ styles.messageBody } >You have { message.status } a meeting with { message.sender.name } { messageEnd }</Text>
      messagePhoto = message.sender.photo
    } else {
      messageText = <Text style={ styles.messageBody } >{ message.receiver.name } has { message.status } a meeting with you { messageEnd }</Text>
      messagePhoto = message.receiver.photo
    }
    return (
      <View style={ styles.messageContainer }>
        <Text style={ styles.date }>{ sentDate }</Text>
        <TouchableOpacity style={ styles.messageInfoContainer } onPress={() => this.setState({showMessage: !this.state.showMessage})}>
          <View style={ styles.imageContainer }>
            <Image source={ {uri: messagePhoto} } style={ styles.userPic }/>
          </View>
          <View style={ styles.messageInfo }>
            <Text style={ styles.name }>{ connection }</Text>
            <Text style={ styles.message }>{ message.status }</Text>
            {
              !this.state.showMessage ? null : messageText
            }
          </View>
        </TouchableOpacity>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(Message)


const styles = StyleSheet.create({
  messageContainer: {
    display: 'flex',
    flexDirection: 'row',
    padding: 20,
    borderBottomWidth: 0.5,
    borderBottomColor: 'gray',
  },
  messageInfoContainer: {
    display: 'flex',
    flexDirection: 'row',
    maxWidth: '100%',
    marginRight: 10
  },
  imageContainer: {
    shadowOffset: {  width: 2,  height: 2 },
    shadowRadius: 4,
    shadowColor: 'black',
    shadowOpacity: .2,
  },
  userPic: {
    height: 50,
    width: 50,
    borderRadius: 25
  },
  date: {
    position: 'absolute',
    right: 20,
    top: 20
  },
  name: {
    fontSize: 18
  },
  message: {
    marginTop: 5,
    color: 'gray',
  },
  messageBody: {
    marginRight: 10
  },
  messageInfo: {
    marginLeft: 10,
    marginRight: 30,
    marginTop: 5
  }
})