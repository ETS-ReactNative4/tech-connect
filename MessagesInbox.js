import React, { Component } from 'react'
import { View, Text, StyleSheet } from 'react-native'
import Message from './Message'
import { getMessages } from './apiCalls'
import { connect } from 'react-redux';
const uuidv1 = require('uuid/v1');

export class MessagesInbox extends Component {
  constructor() {
    super()
    this.state = {
      messages: []
    }
  }

  async componentDidMount() {
    const messages = await getMessages(this.props.api_key)
    this.setState({
      messages
    })
  }

  render() {
    return (
      <View style={ styles.inboxContainer }>
        <View style={ styles.messageContainer }>
        <Text style={ styles.inboxTitle }>Inbox</Text>
        {
          this.state.messages.length ? this.state.messages.map(message => <Message message={message.attributes} key={ uuidv1() } />) : null
        }
        </View>
      </View>
    )
  }
}

export const mapStateToProps = (state) => ({
  api_key: state.user.api_key
})

export default connect(mapStateToProps)(MessagesInbox)

const styles = StyleSheet.create({
  inboxContainer: {
    flex: 1,
    backgroundColor: '#4AA9C5',
    padding: 20,
    paddingTop: 40
  },
  inboxTitle: {
    fontSize: 30,
    marginTop: 5,
    marginBottom: 10,
    color: '#4AA9C5',
    marginLeft: 10
  },
  messageContainer: {
    backgroundColor: '#FFFFFF',
    flex: 1,
  }
})
