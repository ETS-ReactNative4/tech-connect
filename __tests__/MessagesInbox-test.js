import React from 'react'
import { shallow } from 'enzyme'
import { MessagesInbox, mapStateToProps } from '../MessagesInbox'
import Message from '../Message'

describe('MessagesInbox', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<MessagesInbox />)
  })

  it('should match the snapshot', () => {

    expect(wrapper).toMatchSnapshot()
  })
})