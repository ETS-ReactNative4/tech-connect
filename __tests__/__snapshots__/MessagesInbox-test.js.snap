// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesInbox should match the snapshot 1`] = `
<Component
  style={
    Object {
      "backgroundColor": "#4AA9C5",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Component
    style={
      Object {
        "fontSize": 30,
        "marginBottom": 10,
        "marginTop": 10,
      }
    }
  >
    Inbox
  </Component>
  <Component
    style={
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      }
    }
  />
</Component>
`;
